### PR TITLE
[expo] Bump `react-native-screens` and `@react-navigation/*`

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Bumps `react-native-screens` to `4.0.0-beta.11` and `@react-navigation/*` to `v7`.
This PR is a draft because there is a minor issue that has to be resolved by `react-navigation.` I'll include a fix as soon as the new version is published. 

# Test Plan

| bare-expo |  iOS  | Android |
| --------: | :---: | :-----: |
|     Paper | ✅ |  ✅  |
|    Fabric | ✅ |  ✅  |

| Expo Go |  iOS  | Android |
| --------: | :---: | :-----: |
|     Paper | ✅ |  ✅  |

Both apps were tested using `dark` and `light` themes.

### CODE REVIEW

**Review by Kudo — APPROVED** [review]
the code changes look good so far but cli ci are failed. do we need #28109 or it's unrelated?

File: apps/bare-expo/MainNavigator.tsx [hunk 1]
```diff
@@ -92,6 +92,7 @@ function TabNavigator() {
   const { theme } = useTheme();
   return (
     <Tab.Navigator
+      id={undefined}
```
Comment (tsapeta) [inline apps/bare-expo/MainNavigator.tsx:95]: 😫

**Comment by marklawlor** [comment]
@Kudo Happy to coordinate on this.  Right now I'm using `resolutions` to override versions, which will obviously break other packages. When this PR is merged I'll rebase it and merge in https://github.com/expo/expo/pull/28109

**Comment by lukmccall** [comment]
> @lukmccall @Kudo Happy to coordinate on this. Right now, I'm using `resolutions` to override versions, which will obviously break other packages. When this PR is merged I'll rebase it and merge in #28109

I encountered some issues on iOS that needed to be addressed by the navigation team. They have already released a new version. It seems to be working fine, but I am still testing it. If everything goes according to plan, I will merge this pull request today.

**Comment by lukmccall** [comment]
Waiting for  https://github.com/software-mansion/react-native-screens/pull/2404 to be merged and published. Everything else seems to be working fine.

File: apps/native-component-list/src/navigation/StackConfig.tsx [hunk 2]
```diff
@@ -26,13 +26,12 @@ export default function getStackConfig(navigation: BottomTabNavigationProp<any>,
       },
       headerTintColor: theme.icon.info,
       headerTitleStyle: {
-        color: theme.text.default,
+        color: 'theme.text.default',
```
Comment (EvanBacon) [inline apps/native-component-list/src/navigation/StackConfig.tsx:29]: ???

**Comment by fobos531** [comment]
@lukmccall beta-12 was also released: https://github.com/software-mansion/react-native-screens/commit/3ca34864e9d4e67bedef94079516b9d1e5a5b97a

not sure if its important for the context of this PR but just a heads up

File: apps/bare-expo/MainNavigator.tsx (hunk 1 above)
Comment (brentvatne) [reply apps/bare-expo/MainNavigator.tsx:95]: is this needed @satya164?

File: apps/expo-go/src/navigation/defaultNavigationOptions.tsx [hunk 3]
```diff
@@ -19,7 +19,8 @@ export default (theme: ColorTheme): StackNavigationOptions => {
       color: theme === 'dark' ? darkTheme.text.default : lightTheme.text.default,
     },
     headerTintColor: theme === 'dark' ? darkTheme.icon.default : lightTheme.icon.default,
-    headerBackTitleVisible: false,
+    headerBackButtonDisplayMode: 'minimal',
+    headerBackImage: () => <></>,
```
Comment (brentvatne) [inline apps/expo-go/src/navigation/defaultNavigationOptions.tsx:23]: @satya164 is there no equivalent to `headerBackTitleVisible` in react-navigation 7? this feels like a non-obvious way to accomplish the equivalent

**Review by brentvatne — APPROVED** [review]

File: apps/expo-go/src/navigation/defaultNavigationOptions.tsx (hunk 3 above)
Comment (brentvatne) [reply apps/expo-go/src/navigation/defaultNavigationOptions.tsx:23]: @lukmccall - was it intentional to add `headerBackImage: () => <></>` here? it seems like the behavior we had before hid the title, but not the back image.

**Pull request merged**